cogs/welcome/welcome.py

```diff
@@ -29,8 +30,31 @@ def __init__(self, bot: Red):
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
         self.config.register_guild(**DEFAULT_GUILD)
 
-    async def getRandomMessage(self, guild):
-        greetings = await self.config.guild(guild).get_attr(KEY_GREETINGS)()
+    async def getRandomMessage(self, guild: discord.Guild, pool: Optional[GreetingPools] = None):
+        """Gets a random message from a greeting pool.
+
+        If no pool is specified, the default pool is used.
+        If the specified pool is empty, the default pool is used.
+
+        Parameters
+        ----------
+        guild: discord.Guild
+            The guild to get a random greeting from.
+        pool: Optional[GreetingPools]
+            The pool to get a random greeting from.
+        """
+        if pool is None:
+            pool = GreetingPools.DEFAULT
+
+        key = KEY_GREETINGS
+        if pool == GreetingPools.RETURNING:
+            key = KEY_RETURNING_GREETINGS
```

Comment:
Not sure where my previous comment went, so I'll repost.

Can we get rid of the `is None` check? Doesn't seem useful in this case: we can just compare `None` directly with the enum int:
```py
>>> from enum import IntEnum
>>> class TestClass(IntEnum):
...    DEFAULT = 0
...    TEST = 1
>>> None == TestClass.TEST
False
>>> None == TestClass.DEFAULT
False
>>> 1 == TestClass.DEFAULT
False
>>> 0 == TestClass.DEFAULT
True
```
```suggestion
        key = KEY_GREETINGS
        if pool == GreetingPools.RETURNING:
            key = KEY_RETURNING_GREETINGS
```

Reply:
Thanks. Take a look at my new commit.